009-ui-and-admin-server-security.md

```diff
@@ -0,0 +1,107 @@
+The SCRAM-SHA-512 and Mutual TLS support uses the native support in Kafka for authentication and for OAuth2, Strimzi provides an [OAuth2 library](https://github.com/strimzi/strimzi-kafka-oauth) which has a generic OAuth2 interface allowing you to connect to multiple different OAuth2 servers. 
+Browser based UIs do not traditionally use mutual TLS so the proposed mechanisms for authentication on the UI only covers OAuth2 and SCRAM-SHA-512.
+The admin server proposal similarly only covers SCRAM-SHA-512 and OAuth2 support.
+The OAuth2 and SCRAM-SHA-512 mechanisms for the admin server propagates the SCRAM credentials or the OAuth token to Kafka where the Kafka broker performs the authentication.
+Implementing mutual TLS support in the admin server would require a mechanism that could propagate the identity of the client to Kafka as the admin server will be the TLS termination point and authentication would be between the external client and the admin server. Implementing this in a secure way is not trivial and is not included in this proposal.
+
+Both the UI and the admin server will also support a scenario with no security.
+
+### Authorization
+Both security mechanisms allow authorization to be done using the standard Kafka authorizers along with ACL support.
+Alternatives are available with Kafka authorizers being supplied for [KeyCloak](https://www.keycloak.org/) and [Open Policy Agent](https://www.openpolicyagent.org/docs/latest/) as documented in the Strimzi documentation. The proposed authentication mechanisms for the UI and the admin server will allow these documented authorization methods to be used without requiring any additional changes to the documented methods.
+
+## Proposed Authentication Implementations
+### Admin server using SCRAM-SHA-512
+The following describes the flow for authentication and authorization using SCRAM SHA512 as the authentication type when a web client calls a Kafka client API through the admin server.
+In the diagram:
+1. The credentials are encoded in a base64 encoded Basic Auth header and are passed to the admin server.
+2. The admin server decodes the credentials and populates the Kafka client properties JAAS login string with the userid and password.
+3. The Kafka client api is called from the admin server which authenticates the credentials on the Kafka Broker.
+4. If authentication is successful, the principal is passed to a Kafka authorizer for authorization.
+5. If authorization is successful, the command completes and returns the result to the admin server.
+6. The admin server encodes the response in an HTTP response and returns to the web client.
+
+![Strimzi Admin server using SCRAM-SHA512](images/009-scram-sha512-admin-server.png)
+
+The credentials that are used need to be setup through the Strimzi User Operator.
+The admin server is stateless, so the credentials will need to be provided on every request.
```

Comment:
It does imply that an AdminClient will be instantiated on each request. This hasn't been benchmarked and we may need to do that. I think the tradeoff here is going to be the latency vs the complication of managing a session which is the argument for making it stateless. I think it is worth bearing in mind that the API is going to be used for Administration purposes and I don't think it is likely that it will have to deal with large numbers of concurrent requests or high volume.

Reply:
So there are benefits in caching the admin clients in a way described by @mstruk (LRU keyed on the Authorization header value so I will add that in. @mstruk as this is an implementation detail, are you happy for me to describe that in the [Admin Server proposal](https://github.com/strimzi/proposals/pull/9) rather than this proposal which is a more generic description of the security mechanisms?

Reply:
Sure.